packages/wrangler/src/config/validation.ts

```diff
@@ -2561,6 +2562,18 @@ const validateR2Binding: ValidatorFn = (diagnostics, field, value) => {
 		);
 		isValid = false;
 	}
+	if (
+		isValid &&
+		hasProperty(value, "bucket_name") &&
+		!isValidR2BucketName(value.bucket_name)
+	) {
+		diagnostics.errors.push(
+			`"${field}.bucket_name" must begin and end with an alphanumeric, only contain letters (a-z), numbers (0-9), and hyphens (-)," +
+			" and be between 3 and 63 characters long. but got ${JSON.stringify(value.bucket_name)}.`
+		);
```

Comment:
This message is repeated six times in the code. Could it just be added to the `isValidR2BucketName()` function to avoid the duplication?

Reply:
We can do that, probably renaming the function to `validateR2BucketName` returning `{isValid, errorMsg?}`